Ginger/GingerCoreNET/RecordingLib/RecordingManager.cs

```diff
@@ -0,0 +1,198 @@
+                    {
+                        ListPOMObjectHelper.Add(GetRecordingHelperObject(e.PageTitle, e.PageURL, e.ScreenShot));
+                    }
+                    else if (!(CurrentPOM.PageURL == obj.PageURL && CurrentPOM.Name == obj.PageTitle))
+                    {
+                        CurrentPOM = obj.ApplicationPOM;
+                    }
+                }
+                else
+                {
+                    ListPOMObjectHelper = new List<POMObjectRecordingHelper>();
+                    ListPOMObjectHelper.Add(GetRecordingHelperObject(e.PageTitle, e.PageURL, e.ScreenShot));
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error in Page change event handler while recording", ex);
+            }
+        }
+
+        private void PlatformDriver_ElementRecorded(object sender, ElementActionCongifuration e)
+        {
+            try
+            {
+                Act actUI;
+                ElementInfo einfo = null;
+                if (e.LearnedElementInfo != null)
+                {
+                    einfo = (ElementInfo)e.LearnedElementInfo;
+                    e.AddPOMToAction = CreatePOM;
+                    e.POMGuid = CurrentPOM.Guid.ToString();
+                    e.ElementGuid = einfo.Guid.ToString();
+
+                    actUI = PlatformInfo.GetPlatformAction(einfo, e);                    
+                }
+                else
+                {
+                    actUI = PlatformInfo.GetPlatformAction(null, e);
+                }
+                if (actUI != null)
+                {
+                    if (CurrentPOM != null && einfo != null)
+                    {
+                        CurrentPOM.MappedUIElements.Add(einfo);
+                    }
+                    Context.BusinessFlow.AddAct(actUI);
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error in Element recording event handler while recording", ex);
+            }
+        }
+
+        public void StartRecording()
+        {
+            try
+            {
+                PlatformDriver.StartRecording();
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error in Start recording", ex);
+            }
+        }
+
+        public void StopRecording()
+        {
+            try
+            {
+                PlatformDriver.StopRecording();
+                if (ListPOMObjectHelper != null)
+                {
+                    foreach (var cPom in ListPOMObjectHelper)
+                    {
+                        if (!string.IsNullOrEmpty(cPom.PageTitle) && !string.IsNullOrEmpty(cPom.PageURL))
+                        {
+                            try
+                            {
+                                PomLearnUtils utils = new PomLearnUtils(cPom.ApplicationPOM);
```

Comment:
PomLearnUtils is not releated here, please implement Save function inside RecordingManager

Reply:
Done